README.md

```diff
@@ -110,6 +115,94 @@ You can verify your smart contract on Etherscan by running:
 yarn verify --network network_name
 ```
 
+## Hook Example
+
+- useScaffoldContract
+- useScaffoldContractRead
+- useScaffoldContractWrite
+- useScaffoldEventSubscriber
+- useScaffoldEventHistory
+- useDeployedContractInfo
+
+  **useScaffoldContract**:
+  Use to gets a deployed contract by contract name and returns a contract instance
+  Can also be use to read and write to the deployed smart contract
+
+  ```ts
+  const { data: yourContract } = useScaffoldContract({
+    contractName: "YourContract",
+  });
+  // will return the greeting and can be call in any function unlike useScaffoldContractRead
+  await yourContract.greeting();
+
+  //will be use to write to a contract and can be called in any function
+  await yourContract.setGreeting("Hello");
```

Comment:
Please check if this works. The hook uses the provider as default, so I think that you should pass the signer to the hook if you want to call write methods.